### PR TITLE
[REVIEW] Rename Series.unique_count() to nunique() to match pandas API

### LINKED ISSUE #793: [FEA] Rename 'Series.unique_count' to 'nunique' to match Pandas's API
For better compatibility with Pandas, cudf.dataframe.series.unique_count should be deprecated and replaced by an `nunique` method.

It should support the same `dropna` parameter, and support exact counts instead of approximate.

**Describe alternatives you've considered**
Until this is resolved, users can use len(value_counts()) to compute exact count of uniques, but this will be slower.

### PR DESCRIPTION
**Pull request by dillon-cullinan**
Closes #793

### CODE REVIEW

File: python/cudf/tests/test_stats.py [hunk 1]
```diff
@@ -86,14 +86,48 @@ def test_series_unique():
...
         got = sr.value_counts().to_pandas().sort_index()
         print(expect.head())
         print(got.head())
         assert got.equals(expect)
 
+@pytest.mark.parametrize('nan_as_null, dropna',
+                         [(True, True), (True, False),
+                          (False, True), (False, False)])
+#@pytest.mark.parametrize('data', [[1.0, np.nan, np.nan], [1, 2, 3, 2, 2],
+#                                  [1.0, 2.0, 3.0, np.nan, None]])
```
Comment (kkraus14) [inline python/cudf/tests/test_stats.py]: Remove dead code

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,9 +26,10 @@
 - PR #992 Add comparison binaryops to DateTime columns
 - PR #996 Replace relative path imports with absolute paths in tests
 - PR #995 CSV Reader: Add index_col parameter to specify the column name or index to be used as row labels
-- PR #1007 Replace relative path imports with absolute paths in cudf
 - PR #997 Add property index setter
+- PR #1007 Replace relative path imports with absolute paths in cudf
```
Comment (kkraus14) [inline CHANGELOG.md]: This shouldn't be here...

Comment (dillon-cullinan) [reply CHANGELOG.md]: The PR's got out of order on the changelog, not sure. Can revert it if needed.

Comment (kkraus14) [reply CHANGELOG.md]: Ah sorry missed the deletion, this is fine.

File: python/cudf/tests/test_stats.py [hunk 3]
```diff
@@ -95,6 +95,39 @@ def test_series_unique():
...
 
 
+@pytest.mark.parametrize('nan_as_null, dropna',
+                         [(True, True), (True, False),
+                          (False, True), (False, False)])
+def test_series_nunique(nan_as_null, dropna):
+    # We remove nulls as opposed to NaNs using the dropna parameter,
+    # so to test against pandas we replace NaN with another discrete value
+    cudf_series = Series([1, 2, 2, 3, 3], nan_as_null=nan_as_null)
+    pd_series = pd.Series([1 ,2, 2, 3, 3])
```
Comment (kkraus14) [inline python/cudf/tests/test_stats.py]:
```suggestion
    pd_series = pd.Series([1, 2, 2, 3, 3])
```

File: python/cudf/utils/cudautils.py [hunk 4]
```diff
@@ -632,13 +633,23 @@ def run(self, arr, k):
...
 @cuda.jit
-def gpu_mark_segment_begins(arr, markers):
+def gpu_mark_segment_begins_float(arr, markers):
     i = cuda.grid(1)
     if i == 0:
         markers[0] = 1
     elif 0 < i < markers.size:
         if not markers[i]:
-            markers[i] = arr[i] != arr[i - 1]
+            markers[i] = not check_equals_float(arr[i], arr[i - 1])
```
Comment (kkraus14) [inline python/cudf/utils/cudautils.py]:
```suggestion
            markers[i] = not check_equals_float(arr[i], arr[i - 1])
```

File: python/cudf/utils/utils.py [hunk 5]
```diff
@@ -39,6 +40,18 @@ def mask_set(mask, pos):
     mask[pos // mask_bitsize] |= 1 << (pos % mask_bitsize)
 
 
+@njit
+def check_equals_float(a, b):
+    return (a == b or (isnan(a) and isnan(b)) or
+            ((isinf(a) and a < 0) and (isinf(b) and b < 0)) or
+            ((isinf(a) and a > 0) and (isinf(b) and b > 0)))
```
Comment (kkraus14) [inline python/cudf/utils/utils.py]:
```suggestion
            ((isinf(a) and a > 0) and (isinf(b) and b > 0)))
```

**Comment by kkraus14** [comment]
rerun tests

**Pull request merged**